### PR TITLE
[diags] make diag output & cmd line buffer params configurable

### PR DESCRIPTION
**Pull request by bisdash**
Make ot diag output buffer size configurable.
Make ot diag command line max argument number and buffer size configurable.

### CODE REVIEW

File: src/core/openthread-core-default-config.h [hunk 1]
```diff
@@ -1535,4 +1535,34 @@
...
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE
+ *
+ * Define openthread diagnostic mode output buffer size
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE
+#define OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE (256)
```
Comment (pvanhorn) [inline src/core/openthread-core-default-config.h]: while parentheses are typically a good idea in macros, this does not coincide with the style of the file.  Remove here and elsewhere below.

File: src/core/openthread-core-default-config.h [hunk 2]
```diff
@@ -1535,4 +1535,34 @@
 #define OPENTHREAD_CONFIG_DISABLE_CCA_ON_LAST_ATTEMPT 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE
+ *
+ * Define openthread diagnostic mode output buffer size
```
Comment (abtink) [inline src/core/openthread-core-default-config.h]: nit: openthread -> OpenThread. Also "." at the end of lines (here and the ones below).

File: src/core/openthread-core-default-config.h (hunk 1 above)
Comment (bisdash) [reply src/core/openthread-core-default-config.h]: Okay.

File: src/core/openthread-core-default-config.h (hunk 2 above)
Comment (bisdash) [reply src/core/openthread-core-default-config.h]: okay

File: src/core/openthread-core-default-config.h [hunk 3]
```diff
@@ -1535,4 +1535,34 @@
 #define OPENTHREAD_CONFIG_DISABLE_CCA_ON_LAST_ATTEMPT 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE
+ *
+ * Define OpenThread diagnostic mode output buffer size.
```
Comment (jwhui) [inline src/core/openthread-core-default-config.h]: nit: indicate units (i.e. bytes) here and below

**Pull request merged**